Add blog post: LibrePCB 1.0.0 RC1 Released (#68)
LibrePCB/librepcb-website: content/blog/2023-08-21_release_1.0.0-rc1/index.adoc

---
date: 2023-08-21
title: LibrePCB 1.0.0 RC1 Released
author: U. Bruhin
---

> I'm very excited to announce that LibrePCB 1.0 has been completed and
> a first release candidate is available for download!

Reviewer: If it were completed, we'd see a non-RC release, right? You probably mean this:

```suggestion
I'm very excited to announce that LibrePCB 1.0 is feature-complete and that
a first release candidate is available for download!
```

Author: I agree, but are you sure the added "that" is required here? Somehow to me it sounds better without the second "that" :thinking:

Reviewer: To me, it sounds way better. Otherwise, I intuitively start reading the sentence as "I'm very excited to announce that LibrePCB 1.0 is (feature-complete and a first release candidate) is available for download!".

Author: Okay interesting, so I'll take it that way :+1:

About This Release Candidate
----------------------------

LibrePCB 1.0 contains a huge amount of changes compared to the previous
release. Even though we give our best to keep the quality high and do lots

> of testing, such large changes always come with the risk of issues. Thus we

Reviewer: I'd be more direct:

```suggestion
of testing, such large changes always come with the risk of bugs. Thus, we
```

(Or alternatively "problems")

first provide a release candidate prior to the official stable release.

> Generally we do not expect that any serious problems occur with your files
> when using this release candidate. But for critical projects it's a good
> idea anyway to make a backup first (e.g. by using a version control
> system) or to simply wait for the final release.

Reviewer: I think this wording would be better. And in generall, it would be better if people make a backup and test the release candidate, instead of not testing it and then finding bugs once the final release is out. Thus, I'd remove that last part.

```suggestion
Note that while we do not expect that any serious problems occur with your projects
when using this release candidate, it's still a good
idea to create a backup first (e.g. by using a version control
system).
```

  (https://github.com/LibrePCB/LibrePCB/pull/1177[#1177],
  https://github.com/LibrePCB/LibrePCB/pull/1180[#1180])
- *Implement unified production data files generator*
  (https://github.com/LibrePCB/LibrePCB/pull/1194[#1194])
- Support thermal relief pads in planes
  (https://github.com/LibrePCB/LibrePCB/pull/1160[#1160])
- Support blind & buried vias
  (https://github.com/LibrePCB/LibrePCB/pull/1163[#1163])
- Support plated and non-plated slots
  (https://github.com/LibrePCB/LibrePCB/pull/1071[#1071],
  https://github.com/LibrePCB/LibrePCB/pull/1076[#1076])
- Support keepout zones
  (https://github.com/LibrePCB/LibrePCB/pull/1167[#1167])
- Support rounded and custom footprint pad shapes
  (https://github.com/LibrePCB/LibrePCB/pull/1102[#1102],
  https://github.com/LibrePCB/LibrePCB/pull/1103[#1103])
- Support configurable stop mask and solder paste on pads
  (https://github.com/LibrePCB/LibrePCB/pull/1137[#1137])
- Support fiducials and other pad types
  (https://github.com/LibrePCB/LibrePCB/pull/1142[#1142],
  https://github.com/LibrePCB/LibrePCB/pull/1143[#1143])
- And many more!

Next Steps
----------

We expect the final release to be available soon, probably around the

> beginning of september. Meanwhile we will work on migrating our official

Reviewer: ```suggestion
beginning of September. Meanwhile, we will work on migrating our official
```

libraries to the new file format, extending our fabrication service to
support LibrePCB 1.x projects, and updating the documentation accordingly.

---

Download
--------

Since this is not a stable release yet, only portable packages are available

> so far (in the sidebar):

Reviewer: What do you mean with "sidebar"? I don't see a sidebar on the download page, and there are only 0.1.7 links available for the portable packages 🙂

Author: The sidebar will appear once it is released :slightly_smiling_face: 

![image](https://github.com/LibrePCB/librepcb-website/assets/5374821/d938a20b-2263-4e56-8b7e-0a4df6ad99b4)

Reviewer: Ah 👍🏻 Yeah, makes sense. (Maybe the backup recommendation should be in the sidebar as well?)

Author: Alright, I updated the current draft this way:

![image](https://github.com/LibrePCB/librepcb-website/assets/5374821/d8ec3002-2cee-41d9-ac95-919ef7735692)

Reviewer: Looks great! 👍🏻
